units/pod_initializing.go

```diff
@@ -0,0 +1,105 @@
+package units
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/evergreen-ci/cocoa"
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/pod"
+	"github.com/evergreen-ci/utility"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+const podInitJobName = "pod-initializing-job"
+
+func init() {
+	registry.AddJobType(podInitJobName, func() amboy.Job {
+		return makePodInitJob()
+	})
+}
+
+type podInitJob struct {
+	job.Base `bson:"metadata" json:"metadata" yaml:"metadata"`
+	PodID    string `bson:"pod_id" json:"pod_id" yaml:"pod_id"`
+
+	pod       *pod.Pod
+	smClient  cocoa.SecretsManagerClient
+	vault     cocoa.Vault
+	ecsClient cocoa.ECSClient
+	ecsPod    cocoa.ECSPod
+	env       evergreen.Environment
+}
+
+func makePodInitJob() *podInitJob {
+	j := &podInitJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    podInitJobName,
+				Version: 0,
+			},
+		},
+	}
+
+	j.SetDependency(dependency.NewAlways())
+	return j
+}
+
+// NewPodInitJob creates a job that starts the given pod.
+func NewPodInitJob(env evergreen.Environment, p *pod.Pod, id string) amboy.Job {
+	j := makePodInitJob()
+	j.pod = p
+	j.PodID = p.ID
+	j.env = env
+	j.SetPriority(1)
+	j.SetID(fmt.Sprintf("%s.%s.%s", podInitJobName, j.PodID, id))
+	j.SetScopes([]string{fmt.Sprintf("%s.%s", podInitJobName, j.PodID)})
+	j.SetShouldApplyScopesOnEnqueue(true)
+	j.UpdateRetryInfo(amboy.JobRetryOptions{
```

Comment:
You should set a max attempt count to limit the number of times this can retry. It will retry 100 times by default if you don't set it, but that's likely too many - I would set it to maybe 10-20 at the most.